react/src/views/Transfers/TransferAgreementOverview/TransferAgreementOverview.test.tsx

```diff
@@ -0,0 +1,280 @@
+
+  // Data of Mock Transfer is shown correctly
+  expect(await screen.findByRole("cell", { name: /to \/ from/i })).toBeInTheDocument();
+  expect(screen.getByRole("cell", { name: /boxcare/i })).toBeInTheDocument();
+  expect(screen.getByRole("cell", { name: /pending/i })).toBeInTheDocument();
+  expect(screen.getByRole("link", { name: /thessaloniki \(1\)/i })).toBeInTheDocument();
+  expect(screen.getByRole("cell", { name: /Good Comment/i })).toBeInTheDocument();
+  expect(screen.getByRole("cell", { name: /1\/1\/2024/i })).toBeInTheDocument();
+});
+
+const failedMutationTests = [
+  {
+    name: "4.2.3.1a - Accept Transfer Agreement fails due to GraphQLError",
+    mocks: [
+      mockSuccessfullTransferAgreementsQuery({ isInitiator: false }),
+      mockGraphQLError(ACCEPT_TRANSFER_AGREEMENT, { id: "1" }),
+    ],
+    stateButtonText: /request open/i,
+    modalButtonText: /Accept/i,
+    toastText: /could not accept/i,
+  },
+  {
+    name: "4.2.3.1b - Accept Transfer Agreement fails due to NetworkError",
+    mocks: [
+      mockSuccessfullTransferAgreementsQuery({ isInitiator: false }),
+      mockNetworkError(ACCEPT_TRANSFER_AGREEMENT, { id: "1" }),
+    ],
+    stateButtonText: /request open/i,
+    modalButtonText: /Accept/i,
+    toastText: /could not accept/i,
+  },
+  {
+    name: "4.2.4.1a - Reject Transfer Agreement fails due to GraphQLError",
+    mocks: [
+      mockSuccessfullTransferAgreementsQuery({
+        type: TransferAgreementType.SendingTo,
+        isInitiator: false,
+      }),
+      mockGraphQLError(REJECT_TRANSFER_AGREEMENT, { id: "1" }),
+    ],
+    stateButtonText: /request open/i,
+    modalButtonText: /Reject/i,
+    toastText: /could not reject/i,
+  },
+  {
+    name: "4.2.4.1b - Reject Transfer Agreement fails due to NetworkError",
+    mocks: [
+      mockSuccessfullTransferAgreementsQuery({
+        type: TransferAgreementType.ReceivingFrom,
+        isInitiator: false,
+      }),
+      mockNetworkError(REJECT_TRANSFER_AGREEMENT, { id: "1" }),
+    ],
+    stateButtonText: /request open/i,
+    modalButtonText: /Reject/i,
+    toastText: /could not reject/i,
+  },
+  {
+    name: "4.2.5.1a - Cancel Transfer Agreement fails due to GraphQLError",
+    mocks: [
+      mockSuccessfullTransferAgreementsQuery({ state: TransferAgreementState.Accepted }),
+      mockGraphQLError(CANCEL_TRANSFER_AGREEMENT, { id: "1" }),
+    ],
+    stateButtonText: /accepted/i,
+    modalButtonText: /terminate/i,
+    toastText: /could not cancel/i,
+  },
+  {
+    name: "4.2.5.1b - Cancel Transfer Agreement fails due to NetworkError",
+    mocks: [
+      mockSuccessfullTransferAgreementsQuery({ state: TransferAgreementState.Accepted }),
+      mockNetworkError(CANCEL_TRANSFER_AGREEMENT, { id: "1" }),
+    ],
+    stateButtonText: /accepted/i,
+    modalButtonText: /terminate/i,
+    toastText: /could not cancel/i,
+  },
+];
+
+failedMutationTests.forEach(({ name, mocks, stateButtonText, modalButtonText, toastText }) => {
```

Comment:
@vahidbazzaz I serialized some tests here for the first time. It reduces the test code quite a bit. Since we are mostly runnning tests for error and loading states, these tests repeat themselves quite quickly.

Reply:
I know this as parametrization of tests, have used it in the BE in places as well :+1: